jack.py

```diff
@@ -1533,6 +1562,65 @@ def get_array(self):
         return np.frombuffer(self.get_buffer(), dtype=np.float32)
 
 
+def _maybe_bytes(data):
+    if type(data) is bytes:
+        data_ptr = _ffi.new("unsigned char[]", len(data))
+        data_ptr[0:len(data)] = data
+        return data_ptr
+    return data
+
+def _returns_bytes(fun):
+    def wrapper(self, cnt):
+        ret_cnt, data = fun(self, cnt)
+        return bytes(_ffi.buffer(data, ret_cnt))
+    return wrapper
+
+class RingBuffer(object):
+    def __init__(self, size=256):
+        ptr = _lib.jack_ringbuffer_create(size + 1)
+        if not ptr:
+            raise JackError(
+                "Could not create ringbuffer of size {0}".format(size))
+        self._ptr = _ffi.gc(ptr, _lib.jack_ringbuffer_free)
+
+    def _read(self, cnt):
+        data = _ffi.new("unsigned char[]", cnt)
+        ret_cnt = _lib.jack_ringbuffer_read(self._ptr, data, cnt)
+        return ret_cnt, data
+    read = _returns_bytes(_read)
+
+    def _peek(self, cnt):
+        data = _ffi.new("unsigned char[]", cnt)
+        ret_cnt = _lib.jack_ringbuffer_peek(self._ptr, data, cnt)
+        return ret_cnt, data
+    peek = _returns_bytes(_peek)
+
+    def write(self, data):
+        data = _maybe_bytes(data)
+        return _lib.jack_ringbuffer_write(self._ptr, data, _ffi.sizeof(data))
+
+    def mlock(self):
+        return _lib.jack_ringbuffer_mlock(self._ptr)
+
+    def reset(self):
+        _lib.jack_ringbuffer_reset(self._ptr)
+
+    @property
+    def read_space(self):
+        return _lib.jack_ringbuffer_read_space(self._ptr)
+
+    @property
+    def write_space(self):
+        return _lib.jack_ringbuffer_write_space(self._ptr)
+
+    @property
+    def size(self):
+        return self._ptr.size - 1
+
+    def __repr__(self):
+        return "<jack.RingBuffer {0.read_space}/{0.size}>".format(self)
```

Comment:
What about being more explicit?

    def __repr__(self):
        return ("<jack.RingBuffer with {0.read_space}/{0.write_space} bytes " +
                "read/write space>").format(self)

It would look like this:

    <jack.RingBuffer with 11/116 bytes read/write space>

BTW, I'm still not sure if it's a good idea to have this `__repr__()` at all ...

Given all the discussion, the awkwardness and possible confusion, we might be better off just using the default `__repr__()` and do `rb.read_space` and `rb.write_space` if we actually want to know the information?